### PR TITLE
tracee: 0.7.0 -> 0.8.3

### PR DESCRIPTION
**Pull request by 06kellyjac**
###### Description of changes

Updated tracee to 0.8.0. Swapped to using our copy of libbpf since it's now a matching version to the one needed by tracee

After #176152 the tracee build process can now pass in a -target of bpf
without weird overrides

the CFLAG is only needed for the bpf target but it's much easier to just apply it generally, means less maintenance going forward
I might ask upstream for more controllable setting of cflags of individual targets

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/tracee/default.nix [hunk 1]
```diff
@@ -39,34 +28,31 @@ buildGoModule rec {
...
   buildInputs = [ zlib libelf ];
 
   makeFlags = [
     "VERSION=v${version}"
-    "CMD_CLANG_BPF=clang-bpf"
     # don't actually need git but the Makefile checks for it
     "CMD_GIT=echo"
   ];
 
+  NIX_CFLAGS_COMPILE = "-fno-stack-protector";
```
Comment (symphorien) [inline pkgs/tools/security/tracee/default.nix:47]:
```suggestion
  hardeningDisable = [ "stackprotector" ];
```

Comment (06kellyjac) [reply pkgs/tools/security/tracee/default.nix:47]: Thanks. I'll fix that

**Comment by 06kellyjac** [comment]
@ofborg eval

File: pkgs/tools/security/tracee/default.nix [hunk 2]
```diff
@@ -6,67 +6,55 @@
...
+    cp -r ${libbpf}/include/bpf ./dist/libbpf/
+  '';
   buildPhase = ''
     runHook preBuild
     make $makeFlags ''${enableParallelBuilding:+-j$NIX_BUILD_CORES -l$NIX_BUILD_CORES}
     runHook postBuild
   '';
 
+  # tests require a separate go module
+  # integration tests can be ran via ./test.nix in a nixos vm
```
Comment (symphorien) [inline pkgs/tools/security/tracee/default.nix:57]: can you add the test to passthru.tests ?

Comment (06kellyjac) [reply pkgs/tools/security/tracee/default.nix]: done

**Pull request merged**